app/controllers/wpcc/your_results_controller.rb

```diff
@@ -31,10 +40,19 @@ def convert_salary_frequency(salary_frequency)
       Wpcc::SalaryFrequencyConverter.convert(salary_frequency)
     end
 
-    def present(your_results)
-      your_results.map do |period|
+    def present_period_contributions(your_results)
+      your_results.map do |period_contribution|
         Wpcc::PeriodContributionPresenter.new(
-          period,
+          period_contribution,
+          view_context: view_context
+        )
+      end
+    end
```

Comment:
If you make it the public suggestion can we move this lines to the schedule method?